File: pulsar-third/pulsar-chrome-devtools/src/main/java/com/github/kklisura/cdt/protocol/types/network/Response.java
package com.github.kklisura.cdt.protocol.types.network;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2019 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.support.annotations.Optional;
import com.github.kklisura.cdt.protocol.types.security.SecurityState;
import java.util.Map;

/** HTTP response data. */
public class Response {

  private String url;

  private Integer status;

  private String statusText;

  private Map<String, Object> headers;

  @Optional private String headersText;

  private String mimeType;

  @Optional private Map<String, Object> requestHeaders;

  @Optional private String requestHeadersText;

  private Boolean connectionReused;

  private Double connectionId;

  @Optional private String remoteIPAddress;

  @Optional private Integer remotePort;

  @Optional private Boolean fromDiskCache;

  @Optional private Boolean fromServiceWorker;

  @Optional private Boolean fromPrefetchCache;

  private Double encodedDataLength;

  @Optional private ResourceTiming timing;

  @Optional private String protocol;

  private SecurityState securityState;

  @Optional private SecurityDetails securityDetails;

  /** Response URL. This URL can be different from CachedResource.url in case of redirect. */
  public String getUrl() {
    return url;
  }

  /** Response URL. This URL can be different from CachedResource.url in case of redirect. */
  public void setUrl(String url) {
    this.url = url;
  }

  /** HTTP response status code. */
  public Integer getStatus() {
    return status;
  }

  /** HTTP response status code. */
  public void setStatus(Integer status) {
    this.status = status;
  }

  /** HTTP response status text. */
  public String getStatusText() {
    return statusText;
  }

  /** HTTP response status text. */
  public void setStatusText(String statusText) {
    this.statusText = statusText;
  }

  /** HTTP response headers. */
  public Map<String, Object> getHeaders() {
    return headers;
  }

  /** HTTP response headers. */
  public void setHeaders(Map<String, Object> headers) {
    this.headers = headers;
  }

  /** HTTP response headers text. */
  public String getHeadersText() {
    return headersText;
  }

  /** HTTP response headers text. */
  public void setHeadersText(String headersText) {
    this.headersText = headersText;
  }

  /** Resource mimeType as determined by the browser. */
  public String getMimeType() {
    return mimeType;
  }

  /** Resource mimeType as determined by the browser. */
  public void setMimeType(String mimeType) {
    this.mimeType = mimeType;
  }

  /** Refined HTTP request headers that were actually transmitted over the network. */
  public Map<String, Object> getRequestHeaders() {
    return requestHeaders;
  }

  /** Refined HTTP request headers that were actually transmitted over the network. */
  public void setRequestHeaders(Map<String, Object> requestHeaders) {
    this.requestHeaders = requestHeaders;
  }

  /** HTTP request headers text. */
  public String getRequestHeadersText() {
    return requestHeadersText;
  }

  /** HTTP request headers text. */
  public void setRequestHeadersText(String requestHeadersText) {
    this.requestHeadersText = requestHeadersText;
  }

  /** Specifies whether physical connection was actually reused for this request. */
  public Boolean getConnectionReused() {
    return connectionReused;
  }

  /** Specifies whether physical connection was actually reused for this request. */
  public void setConnectionReused(Boolean connectionReused) {
    this.connectionReused = connectionReused;
  }

  /** Physical connection id that was actually used for this request. */
  public Double getConnectionId() {
    return connectionId;
  }

  /** Physical connection id that was actually used for this request. */
  public void setConnectionId(Double connectionId) {
    this.connectionId = connectionId;
  }

  /** Remote IP address. */
  public String getRemoteIPAddress() {
    return remoteIPAddress;
  }

  /** Remote IP address. */
  public void setRemoteIPAddress(String remoteIPAddress) {
    this.remoteIPAddress = remoteIPAddress;
  }

  /** Remote port. */
  public Integer getRemotePort() {
    return remotePort;
  }

  /** Remote port. */
  public void setRemotePort(Integer remotePort) {
    this.remotePort = remotePort;
  }

  /** Specifies that the request was served from the disk cache. */
  public Boolean getFromDiskCache() {
    return fromDiskCache;
  }

  /** Specifies that the request was served from the disk cache. */
  public void setFromDiskCache(Boolean fromDiskCache) {
    this.fromDiskCache = fromDiskCache;
  }

  /** Specifies that the request was served from the ServiceWorker. */
  public Boolean getFromServiceWorker() {
    return fromServiceWorker;
  }

  /** Specifies that the request was served from the ServiceWorker. */
  public void setFromServiceWorker(Boolean fromServiceWorker) {
    this.fromServiceWorker = fromServiceWorker;
  }

  /** Specifies that the request was served from the prefetch cache. */
  public Boolean getFromPrefetchCache() {
    return fromPrefetchCache;
  }

  /** Specifies that the request was served from the prefetch cache. */
  public void setFromPrefetchCache(Boolean fromPrefetchCache) {
    this.fromPrefetchCache = fromPrefetchCache;
  }

  /** Total number of bytes received for this request so far. */
  public Double getEncodedDataLength() {
    return encodedDataLength;
  }

  /** Total number of bytes received for this request so far. */
  public void setEncodedDataLength(Double encodedDataLength) {
    this.encodedDataLength = encodedDataLength;
  }

  /** Timing information for the given request. */
  public ResourceTiming getTiming() {
    return timing;
  }

  /** Timing information for the given request. */
  public void setTiming(ResourceTiming timing) {
    this.timing = timing;
  }

  /** Protocol used to fetch this request. */
  public String getProtocol() {
    return protocol;
  }

  /** Protocol used to fetch this request. */
  public void setProtocol(String protocol) {
    this.protocol = protocol;
  }

  /** Security state of the request resource. */
  public SecurityState getSecurityState() {
    return securityState;
  }

  /** Security state of the request resource. */
  public void setSecurityState(SecurityState securityState) {
    this.securityState = securityState;
  }

  /** Security details for the request. */
  public SecurityDetails getSecurityDetails() {
    return securityDetails;
  }

  /** Security details for the request. */
  public void setSecurityDetails(SecurityDetails securityDetails) {
    this.securityDetails = securityDetails;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/types/network/Response.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.types.network

import ai.platon.cdt.kt.protocol.support.annotations.Optional
import ai.platon.cdt.kt.protocol.types.security.SecurityState
import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Any
import kotlin.Boolean
import kotlin.Double
import kotlin.Int
import kotlin.String
import kotlin.collections.Map

/**
 * HTTP response data.
 */
data class Response(
  @param:JsonProperty("url")
  val url: String,
  @param:JsonProperty("status")
  val status: Int,
  @param:JsonProperty("statusText")
  val statusText: String,
  @param:JsonProperty("headers")
  val headers: Map<String, Any?>,
  @param:JsonProperty("headersText")
  @param:Optional
  val headersText: String? = null,
  @param:JsonProperty("mimeType")
  val mimeType: String,
  @param:JsonProperty("requestHeaders")
  @param:Optional
  val requestHeaders: Map<String, Any?>? = null,
  @param:JsonProperty("requestHeadersText")
  @param:Optional
  val requestHeadersText: String? = null,
  @param:JsonProperty("connectionReused")
  val connectionReused: Boolean,
  @param:JsonProperty("connectionId")
  val connectionId: Double,
  @param:JsonProperty("remoteIPAddress")
  @param:Optional
  val remoteIPAddress: String? = null,
  @param:JsonProperty("remotePort")
  @param:Optional
  val remotePort: Int? = null,
  @param:JsonProperty("fromDiskCache")
  @param:Optional
  val fromDiskCache: Boolean? = null,
  @param:JsonProperty("fromServiceWorker")
  @param:Optional
  val fromServiceWorker: Boolean? = null,
  @param:JsonProperty("fromPrefetchCache")
  @param:Optional
  val fromPrefetchCache: Boolean? = null,
  @param:JsonProperty("encodedDataLength")
  val encodedDataLength: Double,
  @param:JsonProperty("timing")
  @param:Optional
  val timing: ResourceTiming? = null,
  @param:JsonProperty("serviceWorkerResponseSource")
  @param:Optional
  val serviceWorkerResponseSource: ServiceWorkerResponseSource? = null,
  @param:JsonProperty("responseTime")
  @param:Optional
  val responseTime: Double? = null,
  @param:JsonProperty("cacheStorageCacheName")
  @param:Optional
  val cacheStorageCacheName: String? = null,
  @param:JsonProperty("protocol")
  @param:Optional
  val protocol: String? = null,
  @param:JsonProperty("securityState")
  val securityState: SecurityState,
  @param:JsonProperty("securityDetails")
  @param:Optional
  val securityDetails: SecurityDetails? = null,
)
